sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePartitionedUploader.cs

```diff
@@ -261,8 +261,9 @@ internal class DataLakePartitionedUploader
                 // calculate offsets for the next appends, and the final
                 // position to flush
                 long appendedBytes = 0;
-
+#pragma warning disable AZC0107
                 while (enumerator.MoveNextAsync().EnsureCompleted())
```

Comment:
Nit - comment detailing the suppression.

Is this the same as the case in PartitionedUploader?  Could the analyzer possibly recognize this scenario instead of suppressing if it's happening repeatedly?

Reply:
Yes, that's the same scenario: enumerating async enumerable that is actually sync. For that we need to trace the path of the `IAsyncEnumerator` from the method that has created it to the place where it is used. We can create an extension method (something like `EnsureEnumerator`), and make analyzer to check that it is called on the result of the method that has `async` parameter (similarly to the way `EnsureCompleted` is checked). I can create a separate task for that if you ok with this solution (I don't want to include it in this PR, it is too big already).

Reply:
That sounds like a good plan to improve this in the future.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/10109